api/filters/reference_instance_filter.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"context"
+	"fmt"
+	"github.com/Peripli/service-manager/constant"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+const (
+	ReferenceInstanceFilterName        = "ReferenceInstanceFilter"
+	ReferenceParametersAreNotSupported = "Reference service instance parameters are not supported"
+	contextKey                         = "context"
+)
+
+// serviceInstanceVisibilityFilter ensures that the tenant making the provisioning/update request
+// has the necessary visibilities - i.e. that he has the right to consume the requested plan.
+type referenceInstanceFilter struct {
+	repository       storage.TransactionalRepository
+	tenantIdentifier string
+}
+
+func NewReferenceInstanceFilter(repository storage.TransactionalRepository, tenantIdentifier string) *referenceInstanceFilter {
+	return &referenceInstanceFilter{
+		repository:       repository,
+		tenantIdentifier: tenantIdentifier,
+	}
+}
+
+func (*referenceInstanceFilter) Name() string {
+	return ReferenceInstanceFilterName
+}
+
+func (f *referenceInstanceFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
```

Comment:
let use a more meaningful name for the pointer caller

Reply:
Naming the receiver. As we write idiomatic Go code, it's common to use the first letter or a short abbreviation as the name of the receiver. If the name of the struct is Server , we'll usually see s or srv or even server

Reply:
I will just rename it with the full name: "referenceFilter" for example.
The next commit will include it.